### ISSUE TITLE
Refactor MVC to Razor Pages.

### ISSUE THREAD

**Issue by asynts** [codidact/core#57]
The project currently uses the Model-View-Controller (MVC) design pattern. It was decided (citation needed) to use Razor Pages instead.

Please note, that the `src/Auth` project will be moved into a separate repository, therefor, this part of should be refactored later (?).

**Comment by asynts** [comment]
I tried to find a place where this was discussed, but couldn't. The decision was made in November of 2019 I believe. I suspect that a lot of it happened in private messages?

**Comment by ranolfi** [comment]
> Please note, that the `src/Auth` project will be moved into a separate repository, therefor, this part of should be refactored later (?).

Sounds like a plan to me.

> I suspect that a lot of it happened in private messages?

Some of it has, @asynts. The question "why razor pages, and not the mvc pattern?" is [pending](https://discordapp.com/channels/634104110131445811/637322644718616576/674478787928391701) of documentation in the [developer FAQ](https://github.com/codidact/docs/tree/master/Developer-FAQ).  
If one digs deep enough on the old #programming channel (now #programming-general) and in #tech-lead-comms, something useful _may_ come up should dev-docs require it. But other than that, I'll quote what I have easily at hand.

DM with @kieranirving (frontend group leader):

> >     Why Razor pages over MVC pattern
> 
> This was discussed mainly via voice chat, but you can find some references in this post https://discordapp.com/channels/634104110131445811/641360136908046336/668175133055254528 and see the conversation/context.
> 
> The short answer is simple: because it fits better with the general architecture we're going with, in terms of reduced complexity.

Many online resources cover Razor Pages vs MVC in ASP.NET \[Core\]. Two of the most relevant IMO are present in that message from the Discord link above: 

- https://hackernoon.com/asp-net-core-razor-pages-vs-mvc-which-will-create-better-web-apps-in-2018-bd137ae0acaahttps://hackernoon.com/asp-net-core-razor-pages-vs-mvc-which-will-create-better-web-apps-in-2018-bd137ae0acaa
- https://stackify.com/asp-net-razor-pages-vs-mvc/

Additionally:

- https://stackoverflow.com/questions/46777404/why-is-razor-pages-the-recommended-approach-to-create-a-web-ui-in-asp-net-core-2
- https://stackoverflow.com/questions/48121928/asp-net-core-2-0-razor-pages-vs-full-mvc-core

TL;DR: reduced complexity means easier maintainance as the project grows, which is one of our core long-term goals.
